src/main/java/seedu/address/logic/commands/EditModuleCommand.java

```diff
@@ -0,0 +1,173 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE_CODE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE_TITLE;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_MODULES;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.module.Module;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.module.ModuleTitle;
+
+/**
+ * Edits the details of an existing module in Plannit.
+ */
+public class EditModuleCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit-module";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the module identified "
+            + "by the index number used in the displayed module list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_MODULE_CODE + "MODULE_CODE] "
+            + "[" + PREFIX_MODULE_TITLE + "MODULE_TITLE]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_MODULE_CODE + "CS1231S";
+
+    public static final String MESSAGE_EDIT_MODULE_SUCCESS = "Edited Module: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_MODULE = "This module already exists in Plannit.";
+
+    private final Index index;
+    private final EditModuleDescriptor editModuleDescriptor;
+
+    /**
+     * @param index of the module in the filtered module list to edit
+     * @param editModuleDescriptor details to edit the module with
+     */
+    public EditModuleCommand(Index index, EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editModuleDescriptor);
+
+        this.index = index;
+        this.editModuleDescriptor = new EditModuleDescriptor(editModuleDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Module> lastShownList = model.getFilteredModuleList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
```

Comment:
```suggestion
            throw new CommandException(MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
```

@AY2223S1-CS2103T-T10-1/developers Should we import via Messages.blah_blah or import the message as shown in suggested

Reply:
There are already `MESSAGE_SOMETHING` variables declared inside the file. Not explicitly referencing `Messages` class will likely cause confusion.